File: FrontEnd/src/services/paymentService.js
import API from "../api/axiosConfig";

const getPaymentByBookingId = async (bookingId) => {
    const response = await API.get(`/payments/booking/${bookingId}`);
    return response.data;
};

const processPayment = async (bookingId, paymentMethod) => {
    const response = await API.post(`/payments/process/${bookingId}?paymentMethod=${paymentMethod}`);
    return response.data;
};

const verifyPayment = async (paymentId) => {
    const response = await API.get(`/payments/${paymentId}/verify`);
    return response.data;
};

const refundPayment = async (paymentId) => {
    const response = await API.put(`/payments/${paymentId}/refund`);
    return response.data;
};

export default {
    getPaymentByBookingId,
    processPayment,
    verifyPayment,
    refundPayment,
};

File: src/main/java/com/hexaware/fastx/service/PaymentService.java
package com.hexaware.fastx.service;

import com.hexaware.fastx.dto.PaymentDto;

public interface PaymentService {
    PaymentDto getPaymentByBookingId(Long bookingId);
    PaymentDto processPayment(Long bookingId, String paymentMethod);
    PaymentDto verifyPayment(Long paymentId);
    PaymentDto refundPayment(Long paymentId);
    /** Razorpay: record payment with known Razorpay transaction ID. */
    PaymentDto processPaymentWithTransaction(Long bookingId, String paymentMethod, String transactionId);
}
